src/rtc.rs

```diff
@@ -0,0 +1,484 @@
+/*!  Interface to the real time clock */
+
+use crate::rcc::{APB1, BDCR, CR, CSR};
+use crate::stm32::{PWR, RTC};
+use rtcc::{Datelike, Hours, NaiveDate, NaiveDateTime, NaiveTime, Rtcc, Timelike};
+
+/// Invalid input error
+// #[derive(Clone, Copy)]
+#[derive(Debug)]
+pub enum Error {
+    InvalidInputData,
+}
+
+/// RTC clock input source
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum RTCSrc {
+    LSE = 0b01,
+    LSI = 0b10,
+    HSE = 0b11,
+}
+
+/// RTC configuration struct
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct cfgRTC {
+    src: RTCSrc,
+    prediv_s: u16,
+    prediv_a: u8,
+}
+
+/// LSI clock source with prescalers
+/// set clock frequency to 1 hz
+impl Default for cfgRTC {
+    fn default() -> Self {
+        cfgRTC {
+            src: RTCSrc::LSI,
+            prediv_s: 311,
+            prediv_a: 127,
```

Comment:
You discussed this with @Sh3Rm4n before. Like him I think there should be a reference to the relevant manual and section you got these magic numbers from documented.